cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriter.java

```diff
@@ -76,16 +71,11 @@ public DatasetSystemMetadataWriter(MetadataStore metadataStore,
   String[] getSystemTagsToAdd() {
     List<String> tags = new ArrayList<>();
     tags.add(dsInstance.getId());
-    try (SystemDatasetInstantiator dsInstantiator = dsInstantiatorFactory.create();
-         Dataset dataset = dsInstantiator.getDataset(dsInstance)) {
-      if (dataset instanceof RecordScannable) {
-        tags.add(EXPLORE_TAG);
-      }
-      if (dataset instanceof BatchReadable || dataset instanceof BatchWritable) {
-        tags.add(BATCH_TAG);
-      }
-    } catch (IOException e) {
-      throw Throwables.propagate(e);
+    if (dataset instanceof RecordScannable) {
```

Comment:
this seems to be existing code, but why don't we check for ``RecordWritable`` here?